Content.Server/GameObjects/Components/Items/FloorTileItemComponent.cs

```diff
@@ -62,16 +66,27 @@ public void AfterInteract(AfterInteractEventArgs eventArgs)
             var location = eventArgs.ClickLocation.AlignWithClosestGridTile();
             var locationMap = location.ToMap(Owner.EntityManager);
 
-            var desiredTile = (ContentTileDefinition)_tileDefinitionManager[_outputTile];
+            var desiredTile = (ContentTileDefinition)_tileDefinitionManager[_outputTiles[_default]];
 
             if (_mapManager.TryGetGrid(location.GetGridId(Owner.EntityManager), out var mapGrid))
             {
                 var tile = mapGrid.GetTileRef(location);
                 var baseTurf = (ContentTileDefinition)_tileDefinitionManager[tile.Tile.TypeId];
 
-                if (HasBaseTurf(desiredTile, baseTurf.Name) && eventArgs.Target == null && stack.Use(1))
+                if (_outputTiles == null)
                 {
-                    PlaceAt(mapGrid, location, desiredTile.TileId);
+                    return;
+                }
```

Comment:
Why would you snowflake-check one tile first, then run the for loop to check the rest? Just do the for loop and always prioritize the first element of the list.